packages/runtime/test-runtime-utils/src/mocksForRebasing.ts

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuid } from "uuid";
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import {
+	MockContainerRuntime,
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+} from "./mocks";
+
+/**
+ * Specialized implementation of MockContainerRuntime for testing op rebasing, when the runtime will resend
+ * ops to the datastores and all ops within the same batch will have the same sequence number.
+ */
+export class MockContainerRuntimeForRebasing extends MockContainerRuntime {
+	private readonly currentBatch: ITrackableMessage[] = [];
+
+	constructor(
+		dataStoreRuntime: MockFluidDataStoreRuntime,
+		factory: MockContainerRuntimeFactoryForRebasing,
+		overrides?: { minimumSequenceNumber?: number },
+	) {
+		super(dataStoreRuntime, factory, overrides);
+	}
+
+	public process(message: ISequencedDocumentMessage) {
+		// Processing ops will happen in a separate JS turn, so by then, we'd increase
+		// the sequence number and flush the current batch.
+		this.clientSequenceNumber++;
+		this.currentBatch.splice(0);
+
+		super.process(message);
```

Comment:
I'm a bit at a loss RE relationship here between message & this.currentBatch[0], as well as comment about async.
Shouldn't message be coming from this.currentBatch?